optuna_dashboard/_cli.py

```diff
@@ -83,9 +87,72 @@ def auto_select_server(
         return "wsgiref"
 
 
+def create_llm_provider_from_config(config: dict[str, Any]) -> LLMProvider | None:
+    if "llm" not in config:
+        return None
+
+    llm_config = config["llm"]
+
+    if "openai" in llm_config:
+        import optuna_dashboard.llm.openai
+        import openai
+
+        client = openai.OpenAI(**llm_config["openai"]["client"])
+        return optuna_dashboard.llm.openai.OpenAI(
+            client,
+            model=llm_config["openai"]["model"],
+            use_chat_completions_api=llm_config["openai"].get("use_chat_completions_api", False),
+        )
+    elif "azure_openai" in llm_config:
+        import optuna_dashboard.llm.openai
+        import openai
+
+        client = openai.AzureOpenAI(**llm_config["azure_openai"]["client"])
+        return optuna_dashboard.llm.openai.AzureOpenAI(
+            client,
+            model=llm_config["azure_openai"]["model"],
+            use_chat_completions_api=llm_config["azure_openai"].get(
+                "use_chat_completions_api", False
+            ),
+        )
+    else:
+        raise ValueError(
+            "Unsupported LLM provider. Supported providers: 'openai', 'azure_openai'."
+        )
+
+
+def create_artifact_store_from_config(config: dict[str, Any]) -> ArtifactStore | None:
+    if "artifact_store" not in config:
+        return None
+
+    artifact_store: ArtifactStore | None = None
+    if "boto3" in config["artifact_store"]:
+        from optuna.artifacts import Boto3ArtifactStore
+
+        artifact_store = Boto3ArtifactStore(**config["artifact_store"]["boto3"])
+    elif "gcs" in config["artifact_store"]:
+        from optuna.artifacts import GCSArtifactStore
+
+        artifact_store = GCSArtifactStore(**config["artifact_store"]["gcs"])
+    elif "filesystem" in config["artifact_store"]:
+        from optuna.artifacts import FileSystemArtifactStore
+
+        artifact_store = FileSystemArtifactStore(**config["artifact_store"]["filesystem"])
+    else:
+        raise ValueError("Unsupported artifact store configuration.")
+
+    return artifact_store
```

Comment:
It might be good to move these functions to `_config.py` to avoid becoming `_cli.py` large. Then, we can write unit tests like below:

```python
def test_create_artifact_store_from_config() -> None:
    config_content = """
[optuna_dashboard]
storage = "sqlite:///test.db"

[artifact_store.filesystem]
base_path = "/tmp/artifacts"
"""
    with tempfile.NamedTemporaryFile(mode="w", suffix=".toml") as f:
        f.write(config_content)
        f.flush()

        config = load_config_from_toml(f.name)
    artifact_store = create_artifact_store_from_config(config)
    assert isinstance(artifact_store, FileSystemArtifactStore)
```

Reply:
I moved the `create_artifact_store_from_config` and `create_llm_provider_from_config` to `_config.py` and added unit tests for them in the commit https://github.com/optuna/optuna-dashboard/pull/1132/commits/24ad2ba193dcf0bd392c7873326478a2363a2aa9.